packages/fetchai/skills/erc1155_client/handlers.py

```diff
@@ -0,0 +1,237 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This package contains handlers for the generic buyer skill."""
+
+from typing import List, Optional, cast
+
+from aea.configurations.base import ProtocolId
+from aea.decision_maker.messages.transaction import TransactionMessage
+from aea.protocols.base import Message
+from aea.skills.base import Handler
+
+from packages.fetchai.protocols.fipa.message import FIPAMessage
+from packages.fetchai.protocols.fipa.serialization import FIPASerializer
+from packages.fetchai.protocols.oef.message import OEFMessage
+from packages.fetchai.skills.erc1155_client.strategy import Strategy
+
+
+class FIPAHandler(Handler):
+    """This class implements a FIPA handler."""
+
+    SUPPORTED_PROTOCOL = FIPAMessage.protocol_id  # type: Optional[ProtocolId]
+
+    def setup(self) -> None:
+        """
+        Implement the setup.
+
+        :return: None
+        """
+        pass
+
+    def handle(self, message: Message) -> None:
+        """
+        Implement the reaction to a message.
+
+        :param message: the message
+        :return: None
+        """
+        # convenience representations
+        fipa_msg = cast(FIPAMessage, message)
+
+        # handle message
+        if fipa_msg.performative == FIPAMessage.Performative.INFORM:
+            self._handle_inform(fipa_msg)
+
+    def teardown(self) -> None:
+        """
+        Implement the handler teardown.
+
+        :return: None
+        """
+        pass
+
+    def _handle_inform(self, msg: FIPAMessage) -> None:
+        """
+        Handle the match inform.
+
+        :param msg: the message
+        :return: None
+        """
+        self.context.logger.info(
+            "[{}]: received INFORM from sender={}".format(
+                self.context.agent_name, msg.counterparty[-5:]
+            )
+        )
+        if len(msg.info.keys()) >= 1:
+            data = msg.info
+            if "contract" in data.keys():
+                contract = self.context.contracts.erc1155
+                contract.set_instance(
```

Comment:
This usage suggests it would be good to also expose an abstract method on the base contract which performs `set_instance` and `set_address` in one; i.e. `set_instance_w_address`.

Reply:
Yes. That's true. All the set_address does is actually reinstantiate the contract. So I don't think that we should introduce a new function. Just remove the assert from the set_address? Thoughts?

Reply:
I ll add the third function ass we discussed.